packages/shared-state-babel_links_info/files/usr/bin/shared-state-publish_babel_links_info

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/lua
+
+-- ! LibreMesh
+-- ! Copyright (c) 2023  Javier Jorge <jjorge@inti.gob.ar>
+-- ! Copyright (c) 2023  Instituto Nacional de Tecnología Industrial
+-- ! Copyright (C) 2023  Asociación Civil Altermundi <info@altermundi.net>
+-- !
+-- ! This program is free software: you can redistribute it and/or modify
+-- ! it under the terms of the GNU Affero General Public License as
+-- ! published by the Free Software Foundation, either version 3 of the
+-- ! License, or (at your option) any later version.
+-- !
+-- ! This program is distributed in the hope that it will be useful,
+-- ! but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- ! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- ! GNU Affero General Public License for more details.
+-- !
+-- ! You should have received a copy of the GNU Affero General Public License
+-- ! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+local JSON = require("luci.jsonc")
+local utils = require('lime.utils')
+local ubus = require "ubus"
+
+local ifaceip = {}
```

Comment:
Why is this allocated outside the function?

Reply:
The idea is that each time the script is run this structure contains an empty list witch will be filled with each call to the get_ip method. Subsequent calls will use the information contained in the list.